java/client/src/main/java/glide/api/models/BaseTransaction.java

```diff
@@ -0,0 +1,188 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models;
+
+import static redis_request.RedisRequestOuterClass.RequestType.CustomCommand;
+import static redis_request.RedisRequestOuterClass.RequestType.GetString;
+import static redis_request.RedisRequestOuterClass.RequestType.Info;
+import static redis_request.RedisRequestOuterClass.RequestType.Ping;
+import static redis_request.RedisRequestOuterClass.RequestType.SetString;
+
+import glide.api.models.commands.InfoOptions;
+import glide.api.models.commands.InfoOptions.Section;
+import glide.api.models.commands.SetOptions;
+import glide.api.models.commands.SetOptions.ConditionalSet;
+import glide.api.models.commands.SetOptions.SetOptionsBuilder;
+import lombok.Getter;
+import org.apache.commons.lang3.ArrayUtils;
+import redis_request.RedisRequestOuterClass.Command;
+import redis_request.RedisRequestOuterClass.Command.ArgsArray;
+import redis_request.RedisRequestOuterClass.RequestType;
+import redis_request.RedisRequestOuterClass.Transaction;
+
+/**
+ * Base class encompassing shared commands for both standalone and cluster mode implementations in a
+ * transaction. Transactions allow the execution of a group of commands in a single step.
+ *
+ * <p>Command Response: An array of command responses is returned by the client exec command, in the
+ * order they were given. Each element in the array represents a command given to the transaction.
+ * The response for each command depends on the executed Redis command. Specific response types are
+ * documented alongside each method.
+ *
+ * @param <T> child typing for chaining method calls
+ */
+@Getter
+public abstract class BaseTransaction<T extends BaseTransaction<T>> {
+    /** Command class to send a single request to Redis. */
+    protected final Transaction.Builder transactionBuilder = Transaction.newBuilder();
+
+    protected abstract T getThis();
```

Comment:
minor - instead of using inheritance & virtual calls, you can just implement this here and cast `this` to T.

Reply:
or you can do 
```java
protected T getThis() {
  return (T) this;
}
```
and suppress/handle the compiler warning once.

but this is a minor point, do whatever you see fit.